tests/utils.py

```diff
@@ -386,71 +423,135 @@ def load_and_filter(file_path):
 
     return np.array_equal(np.sort(events1, axis=0), np.sort(events2, axis=0))
 
-class ReportEntry:
-    __slots__ = (
-        "report_name", "target_name", "report_type",
-        "report_variable", "unit", "report_format",
-        "target_type", "dt", "start_time", "end_time",
-        "num_gids", "buffer_size", "gids",
-    )
-
-    def __init__(self, tokens):
-        self.report_name = tokens[0]
-        self.target_name = tokens[1]
-        self.report_type = tokens[2]
-        self.report_variable = tokens[3]
-        self.unit = tokens[4]
-        self.report_format = tokens[5]
-        self.target_type = int(tokens[6])
-        self.dt = float(tokens[7])
-        self.start_time = float(tokens[8])
-        self.end_time = float(tokens[9])
-        self.num_gids = int(tokens[10])
-        self.buffer_size = int(tokens[11])
-        self.gids = []
+class ReportReader:
+    def __init__(self, file: str):
+        self._reader = ElementReportReader(file)
+        self.populations: Dict[str, Tuple[List[int], pd.DataFrame]] = {}
+
+        for name in sorted(self._reader.get_population_names()):
+            pop = self._reader[name]
+            node_ids = sorted(pop.get_node_ids())
+            data = pop.get()
+
+            # stable sort for ties
+            df = pd.DataFrame(
+                data.data,
+                columns=pd.MultiIndex.from_arrays(data.ids.T),
+                index=data.times
+            ).sort_index(axis=1)
+
+            self.populations[name] = (node_ids, df)
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, ReportReader):
+            return NotImplemented
+
+        if set(self.populations.keys()) != set(other.populations.keys()):
+            return False
+
+        for name in self.populations:
+            nodes1, df1 = self.populations[name]
+            nodes2, df2 = other.populations[name]
+
+            if nodes1 != nodes2:
+                return False
+
+            # coreneuron has sometimes garbage for the first line
+            # erro thresholds as for old bb5 itegration report tests
+            if not np.allclose(df1.values[1:], df2.values[1:], rtol=1e-6, atol=1e-6):
```

Comment:
One more comment, can we cross-check the tolerance with the bb5 one https://github.com/BlueBrain/blueconfigs/blob/8fb55967180357047a11d5f55aa173fdca8bd325/ci/compare_sonata_reports.py#L72 ?

Reply:
I did what I could. I am setting an accuracy level of 1e-16 wherever possible. The old tests have indeed 1e-16. However, for the new ones there are 2 cases where I need to increate them:
- **coreneuron vs neuron reference**: results are not exactly the same
- **summation vs compartment**: I need to perform a summation after report printing. There are truncating errors that I cannot avoid

I let you check the results